Module1.TypesAndClasses/Module1.TypesAndClasses/Shapes/AbstractShape.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Module1.TypesAndClasses.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Module1.TypesAndClasses.Shapes
+{
+    public abstract class AbstractShape
```

Comment:
let's remove it and use BaseShape everywhere, if the method is needed - move it to BaseShape as well